tiledb/sm/array_schema/array_schema.h

```diff
@@ -472,7 +533,7 @@ class ArraySchema {
    * Returns false if the union of attribute and dimension names contain
    * duplicates.
    */
-  bool check_attribute_dimension_names() const;
+  Status check_attribute_dimension_label_names() const;
```

Comment:
Consider changing to throw instead of Status.

Reply:
I'm leaving this change until after `ArraySchema::check()` is deprecated since we want a Status and not exception there.